nntrainer/src/neuralnet.cpp

```diff
@@ -431,15 +431,15 @@ int NeuralNetwork::init() {
       NN_INI_RETURN_STATUS();
       break;
     }
-    previous_dim = layers.back()->getOutputDimension();
-
     const char *acti_str = iniparser_getstring(
       ini, (layer_name + ":Activation").c_str(), unknown);
     ActiType act = (ActiType)parseType(acti_str, TOKEN_ACTI);
 
     layers.back()->setActivation(act);
     status = initActivationLayer(act);
     NN_INI_RETURN_STATUS();
+
```

Comment:
Some layers do not need the activation such as pooling, flatten. No need to check every layer.